index/server/registry-REST-API.adoc

```diff
@@ -0,0 +1,2088 @@
+      "Quarkus"
+    ],
+    "icon": "https://design.jboss.org/quarkus/logo/final/SVG/quarkus_icon_rgb_default.svg",
+    "projectType": "quarkus",
+    "language": "java",
+    "versions": [
+      {
+        "version": "1.1.0",
+        "schemaVersion": "2.2.0",
+        "default": true,
+        "git": {
+          "remotes": {
+            "origin": "https://github.com/devfile-samples/devfile-sample-code-with-quarkus.git"
+          }
+        },
+        "description": "java quarkus with devfile v2.2.0"
+      }
+    ]
+  }
+]
+----
+
+== Gets registry stack devfile
+Gets the specific registry stack devfile content from HTTP response
+
+Note: this REST API only returns the content of `devfile.yaml`, it won't return other resources in the stack
+
+=== HTTP request
+```
+GET http://{registry host}/devfiles/{stack}
+```
+
+=== Request parameters
+[cols="1,1"]
+|===
+|Parameter|Description
+
+|Registry host
+|The URL/ingress that exposes registry service
+
+|Stack
+|Registry stack name
+
+|===
+
+=== Request body
+The request body must be empty.
+=
+== Request example
+```
+curl http://devfile-registry.192.168.1.1.nip.io/devfiles/nodejs
+```
+
+=== Response example
+[source,yaml]
+----
+schemaVersion: 2.0.0
+metadata:
+  name: nodejs
+  version: 1.0.0
+starterProjects:
+  - name: nodejs-starter
+    git:
+      remotes:
+        origin: "https://github.com/odo-devfiles/nodejs-ex.git"
+components:
+  - name: runtime
+    container:
+      image: registry.access.redhat.com/ubi8/nodejs-12:1-45
+      memoryLimit: 1024Mi
+      mountSources: true
+      sourceMapping: /project
+      endpoints:
+        - name: http-3000
+          targetPort: 3000
+commands:
+  - id: install
+    exec:
+      component: runtime
+      commandLine: npm install
+      workingDir: /project
+      group:
+        kind: build
+        isDefault: true
+  - id: run
+    exec:
+      component: runtime
+      commandLine: npm start
+      workingDir: /project
+      group:
+        kind: run
+        isDefault: true
+  - id: debug
+    exec:
+      component: runtime
+      commandLine: npm run debug
+      workingDir: /project
+      group:
+        kind: debug
+        isDefault: true
+  - id: test
+    exec:
+      component: runtime
+      commandLine: npm test
+      workingDir: /project
+      group:
+        kind: test
+        isDefault: true
+----
+
+== Gets registry stack devfile with version
+Gets the specific registry stack devfile content with stack version from HTTP response
+
+Note: this REST API only returns the content of `devfile.yaml`, it won't return other resources in the stack
+
+=== HTTP request
+```
+GET http://{registry host}/devfiles/{stack}/{version}
+```
+
+=== Request parameters
+[cols="1,1"]
+|===
+|Parameter|Description
+
+|Registry host
+|The URL/ingress that exposes registry service
+
+|Stack
+|Registry stack name
+
+|Version
+|Specific version of the stack, or `latest`
+
+|===
+
+=== Request body
+The request body must be empty.
+=
```

Comment:
Remove, `Request Example` subheading is not formatted properly

Reply:
updated